src/lib/connectors/shared/async-html.ts

```diff
@@ -48,22 +53,43 @@ export class AsyncHTMLDocument implements IAsyncHTMLDocument {
             return [];
         }
 
-        return nodeIds.map((nodeId) => {
-            return new AsyncHTMLElement(this._nodes.get(nodeId), this, this._DOM); // eslint-disable-line no-use-before-define, typescript/no-use-before-define
-        });
+        const nodes: Array<AsyncHTMLElement> = [];
+
+        for (let i = 0; i < nodeIds.length; i++) {
+            const nodeId = nodeIds[i];
+            const node = this._nodes.get(nodeId);
+
+            if (node) {
+                nodes.push(new AsyncHTMLElement(node, this, this._DOM)); // eslint-disable-line no-use-before-define, typescript/no-use-before-define
+            } else {
+                // This node was added in the DOM and we don't have it cached
+                // TODO: uncomment once chrome 62 is released.
+                // See https://github.com/cyrus-and/chrome-remote-interface/issues/255 for more information
+                // node = await this._DOM.describeNode({ nodeId });
+                // this._nodes.set(nodeId, node);
+            }
+        }
+
+        return nodes;
     }
 
     public async pageHTML(): Promise<string> {
+        if (this._outerHTML) {
```

Comment:
@molant I think we talk about this some time ago, but we can't cache the outherHTML, because it can change over time.

Reply:
After talking with @sarvaje the new strategy is basically:
> The connector has waited enough time and sonar will scan the site as it is at that moment. If something changes while we are doing the analysis the user should increase the `waitFor` value.